### PR TITLE
feat: force enable socket CLI flag

### PR DESCRIPTION
**Pull request by mgaitan**
I need to have sockets disabled by default in pyproject.yml but still be able to override it from the command line. 

so `pytest --force-enable-socket` takes precedence even when pyproject.yml has

```
addopts = "--disable-socket"
```

### CODE REVIEW

**Comment by miketheman** [comment]
Hi @mgaitan !
I don't think this approach is inline with the utility of the plugin.
It seems like you're looking to effectively disable the plugin via command line.

This behavior exists in `pytest` already - see https://docs.pytest.org/en/stable/how-to/plugins.html#deactivating-unregistering-a-plugin-by-name

Have you tried your scenario with `pytest -p no:socket ...` ?

**Comment by mgaitan** [comment]
disabling the plugin causes the default arguments I already have in pyproject.toml to be unknown and it cannot be executed.

```
[core] src (local)# pytest -p no:socket shiphero_app/tests/core/utils/test_sqs.py::test_msg_dedup_id_cleanup
ENV: local
DEBUG: True
DEBUGGER: False
BASE_DIR: /src/shiphero_app
PYTHONPATH: :/src:/src/shiphero_app
PYTHONUNBUFFERED: 0
READ_DB: mysql://root:****@mysql/mydb
MASTER_DB: mysql://root:****@mysql/mydb
MONGODB: shiphero-mongo:27017
REDSHIFT: redshift
LOG_GROUP: /shiphero/dev/catchall/logs
TIER:
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --disable-socket --allow-unix-socket --allow-hosts=172.18.0.0/24
  inifile: /src/pyproject.toml
  rootdir: /src
```

**Comment by miketheman** [comment]
@mgaitan Thanks for describing your use case. It sounds like the way I'd handle this would be to remove the socket parameters from the config file, and selectively add the `--disable-socket` params when needed, and thus have the control you're looking for.

**Comment by mgaitan** [comment]
@miketheman Please reconsider this. This is not a strange pattern in command line applications at all. 

Actually, many applications honor an environment variable for an argument, but disregard the environment variable in case the argument is explicitly passed. 

To give you a closer and more concrete example related to pytest. In our suite we use [pytest-vcr](https://pytest-vcr.readthedocs.io/en/latest/) [1] to "cache" HTTP interactions of our tests and avoid relying on the internet. In our `pyproject.toml` we have the config

```init
[tool.pytest.ini_options]
addopts = """\
    -v --vcr-record=none \
    ... 
```
which is the [recommended way](https://pytest-vcr.readthedocs.io/en/latest/configuration/#-vcr-record) to make CI fails if a test is making a request that is not considered in the cassette. 

However, when we need to record a new cassette (or update an existing one) this setting should be "overloaded'' and is simply done by passing `--vcr-record=all` explicitly in the local execution of pytest, without needing to change the config from `pyproject.toml`

So basically, `pyproject.toml` config defines the defaults for the project but any explicit cli arguments takes precedence. 


[1] Actually this is the reason why I needed to implementd this change. As we are using  both pytest-vcr and pytest-socket, whenever I need to record a new cassette I need to allow that internet access. So I want to override both the socket disabling and vcr-record=none on that single pytest execution from CLI.

**Comment by miketheman** [comment]
@mgaitan Thanks for the deeper explanation - that use case makes a lot more sense. That context might have been useful at the outset to prevent the back and forth. 😄 

In this case, I think the CLI arg should reflect what is actually happening, so possibly `--override-disable-socket` is clearer, as the `enable-socket` concept is already used for the pytest fixture/decorators.

**Comment by mgaitan** [comment]
@miketheman I picked `--force-enable-socket` to make it even more explicit. I hope you are happy with it.

**Review by miketheman — APPROVED** [review]
Thanks for the updates!
